api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -269,19 +273,25 @@ export class ScenariosService {
     scenarioId: string,
     userId: string,
     input: UpdateScenarioDTO,
-  ): Promise<Either<typeof forbiddenError | typeof lockedScenario, Scenario>> {
+  ): Promise<
+    Either<
+      typeof forbiddenError | typeof lockedByAnotherUser | typeof noLockInPlace,
+      Scenario
+    >
+  > {
     await this.assertScenario(scenarioId);
     const scenario = await this.getById(scenarioId, {
       authenticatedUser: { id: userId },
     });
     if (isLeft(scenario)) {
       return left(forbiddenError);
     }
-    if (!(await this.scenarioAclService.canEditScenario(userId, scenarioId))) {
-      return left(forbiddenError);
-    }
-    if (!(await this.lockService.isLockedByUser(scenarioId, userId))) {
-      return left(lockedScenario);
+    const aclAndLockLogic = await this.scenarioAclService.canEditScenarioAndLockLogicIsCorrect(
```

Comment:
see my earlier nitpick about the name of the function we call here - to combine nitpicking with nitpicking, I suggest to simplify the naming here as all we want to know is whether we can go ahead with the requested update... maybe
```suggestion
    const userCanEditScenario = await this.scenarioAclService.canEditScenarioAndLockLogicIsCorrect(
```